src/libs/actions/Report.js

```diff
@@ -327,34 +327,42 @@ function openReport(reportID, participantList = [], newReportObject = {}) {
         }],
     };
 
+    const params = {
+        reportID,
+        emailList: participantList ? participantList.join(',') : '',
+    };
+
     // If we are creating a new report, we need to add the optimistic report data and a report action
     if (!_.isEmpty(newReportObject)) {
-        onyxData.optimisticData[0].value = {
-            ...onyxData.optimisticData[0].value,
-            ...newReportObject,
-            pendingFields: {
-                createChat: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
-            },
-            isOptimisticReport: true,
-        };
-
         // Change the method to set for new reports because it doesn't exist yet, is faster,
         // and we need the data to be available when we navigate to the chat page
         onyxData.optimisticData[0].onyxMethod = CONST.ONYX.METHOD.SET;
 
         // Also create a report action so that the page isn't endlessly loading
+        const optimisticReportAction = ReportUtils.buildOptimisticCreatedReportAction(newReportObject.ownerEmail);
         onyxData.optimisticData[1] = {
             onyxMethod: CONST.ONYX.METHOD.SET,
             key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
-            value: ReportUtils.buildOptimisticCreatedReportAction(newReportObject.ownerEmail),
+            value: optimisticReportAction,
         };
+
+        onyxData.optimisticData[0].value = {
+            ...onyxData.optimisticData[0].value,
+            ...newReportObject,
+            pendingFields: {
+                createChat: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+            },
+
+            // We add 1ms such that the newly created report appears as read.
```

Comment:
Had to do this because sometimes it's the same timestamp, and we have strictly less than here: https://github.com/Expensify/App/blob/a49a1726dfbdc903fc065a32df7d8b1f598d4cab/src/libs/ReportUtils.js#L1022

Unsure if this could be modified to "less than or equal", I think as is it makes more sense.

Reply:
I was thinking we could update the check to less than or equal

and then also use the same created timestamp to pass in to lastReadTime and the lastActionCreated so that it would always be marked as read instead of being milliseconds off

Reply:
Yeah I think this should be a less than or equal to check, but let's ask @puneetlath since he wrote it.

Reply:
Hm, I think we should leave it as less than. Often we set the `lastReadTime` and `lastCreatedAction` as the same time so that the chat will be read. For example [here](https://github.com/Expensify/App/blob/4ad25a074da2dac76a62565bb878cbaf3ca62840/src/libs/actions/Report.js#L212-L219). 

In this scenario can we not do that?

Reply:
Added this to set them at the same time